src/Authentication/GraphPhpLeagueAccessTokenProvider.php

```diff
@@ -38,13 +38,16 @@ class GraphPhpLeagueAccessTokenProvider extends PhpLeagueAccessTokenProvider
      * @param array<string> $scopes if left empty, it's set to ["https://[graph national cloud host]/.default"] scope
      * @param string $nationalCloud Defaults to https://graph.microsoft.com. See
      * https://learn.microsoft.com/en-us/graph/deployments
+     * @param AccessTokenCache|null $accessTokenCache Defaults to an in-memory cache if null
      */
     public function __construct(
         TokenRequestContext $tokenRequestContext,
         array $scopes = [],
-        string $nationalCloud = NationalCloud::GLOBAL
+        string $nationalCloud = NationalCloud::GLOBAL,
+        ?AccessTokenCache $accessTokenCache = null
```

Comment:
should we set the default value to our implementation instead?

Reply:
This is set at the Kiota core level in the [auth package](https://github.com/microsoft/kiota-authentication-phpleague-php/blob/dev/src/PhpLeagueAccessTokenProvider.php#L95)